crates/bevy_pbr/src/atmosphere/mod.rs

```diff
@@ -0,0 +1,455 @@
+            .add_systems(
+                Render,
+                (
+                    configure_camera_depth_usages.in_set(RenderSet::ManageViews),
+                    queue_render_sky_pipelines.in_set(RenderSet::Queue),
+                    prepare_atmosphere_textures.in_set(RenderSet::PrepareResources),
+                    prepare_atmosphere_transforms.in_set(RenderSet::PrepareResources),
+                    prepare_atmosphere_bind_groups.in_set(RenderSet::PrepareBindGroups),
+                ),
+            )
+            .add_render_graph_node::<ViewNodeRunner<AtmosphereLutsNode>>(
+                Core3d,
+                AtmosphereNode::RenderLuts,
+            )
+            .add_render_graph_edges(
+                Core3d,
+                (
+                    // END_PRE_PASSES -> RENDER_LUTS -> MAIN_PASS
+                    Node3d::EndPrepasses,
+                    AtmosphereNode::RenderLuts,
+                    Node3d::StartMainPass,
+                ),
+            )
+            .add_render_graph_node::<ViewNodeRunner<RenderSkyNode>>(
+                Core3d,
+                AtmosphereNode::RenderSky,
+            )
+            .add_render_graph_edges(
+                Core3d,
+                (
+                    Node3d::MainOpaquePass,
+                    AtmosphereNode::RenderSky,
+                    Node3d::MainTransparentPass,
+                ),
+            );
+    }
+}
+
+/// This component describes the atmosphere of a planet. Most atmospheric
+/// particles scatter and absorb light in two main ways:
+///
+/// Rayleigh scattering occurs among very small particles, like individual gas
+/// molecules. It's wavelength dependent, and causes colors to separate out as
+/// light travels through the atmosphere. These particles *don't* absorb light.
+///
+/// Mie scattering occurs among slightly larger particles, like dust and sea spray.
+/// These particles *do* absorb light, but Mie scattering and absorption is
+/// *wavelength independent*.
+///
+/// Ozone acts differently from the other two, and is special-cased because
+/// it's very important to the look of Earth's atmosphere. It's wavelength
+/// dependent, but only *absorbs* light. Also, while the density of particles
+/// participating in Rayleigh and Mie scattering falls off roughly exponentially
+/// from the planet's surface, ozone only exists in a band centered at a fairly
+/// high altitude.
+///
+/// Note that all units are in kilometers. This is to combat precision issues,
+/// since integrating very small atmospheric densities over long distances
+/// might otherwise cause problems. [`AtmosphereSettings`] has a field to set
+/// the conversion factor from scene units to km, which by default assumes that
+/// the scene units are meters.
+#[derive(Clone, Component, Reflect, ShaderType)]
+#[require(AtmosphereSettings)]
+pub struct Atmosphere {
+    /// Radius of the planet
+    ///
+    /// units: km
```

Comment:
While it might be beneficial for the computations to be done in km, the public API should use meters to stay consistent with the rest of bevys units. This conversion precision loss will be a one time thing when computing the uniform, and will be completely negligible.